test/e2e/singlecluster/pytorchjob_test.go

```diff
@@ -0,0 +1,156 @@
+				Request(kftraining.PyTorchJobReplicaTypeMaster, corev1.ResourceCPU, "1").
+				Request(kftraining.PyTorchJobReplicaTypeMaster, corev1.ResourceMemory, "200Mi").
+				Image(kftraining.PyTorchJobReplicaTypeWorker, util.E2eTestAgnHostImage, util.BehaviorWaitForDeletion).
+				Request(kftraining.PyTorchJobReplicaTypeWorker, corev1.ResourceCPU, "1").
+				Request(kftraining.PyTorchJobReplicaTypeWorker, corev1.ResourceMemory, "200Mi").
+				Obj()
+
+			ginkgo.By("Create a PyTorch", func() {
+				util.MustCreate(ctx, k8sClient, pytorch)
+			})
+
+			ginkgo.By("Waiting for replicas to be ready", func() {
+				createdPyTorch := &kftraining.PyTorchJob{}
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					// Fetch the PyTorch object
+					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(pytorch), createdPyTorch)
+					g.Expect(err).ToNot(gomega.HaveOccurred(), "Failed to fetch PyTorch")
+
+					// Check the worker replica status exists
+					masterReplicaStatus, ok := createdPyTorch.Status.ReplicaStatuses[kftraining.PyTorchJobReplicaTypeMaster]
+					g.Expect(ok).To(gomega.BeTrue(), "Master replica status not found in PyTorch status")
+					workerReplicaStatus, ok := createdPyTorch.Status.ReplicaStatuses[kftraining.PyTorchJobReplicaTypeWorker]
+					g.Expect(ok).To(gomega.BeTrue(), "Worker replica status not found in PyTorch status")
+
+					// Check the number of active replicas
+					g.Expect(masterReplicaStatus.Active).To(gomega.Equal(int32(1)), fmt.Sprintf("Unexpected number of active %s replicas", kftraining.PyTorchJobReplicaTypeMaster))
+					g.Expect(workerReplicaStatus.Active).To(gomega.Equal(int32(2)), fmt.Sprintf("Unexpected number of active %s replicas", kftraining.PyTorchJobReplicaTypeWorker))
+
+					// Ensure PyTorch job has "Running" condition with status "True"
+					runningConditionFound := false
+					for _, condition := range createdPyTorch.Status.Conditions {
+						if condition.Type == kftraining.JobRunning && condition.Status == corev1.ConditionTrue {
+							runningConditionFound = true
+							break
+						}
+					}
+					g.Expect(runningConditionFound).To(gomega.BeTrue(), "Running condition with status True not found")
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			wlLookupKey := types.NamespacedName{
+				Name:      pytorchjob.GetWorkloadNameForPyTorchJob(pytorch.Name, pytorch.UID),
+				Namespace: ns.Name,
+			}
+			createdWorkload := &kueue.Workload{}
+			ginkgo.By("Check workload is created", func() {
+				gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+			})
+
+			ginkgo.By("Check workload is admitted", func() {
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, createdWorkload)
+
+				// Wait for active pods and terminate them
+				listOpts := client.InNamespace(ns.Name)
+				util.WaitForActivePodsAndTerminate(ctx, k8sClient, restClient, cfg, ns.Name, 3, 0, listOpts)
```

Comment:
What's the purpose of the termination? Can we check whether the workload has finished?